### PR TITLE
RKE2 K8s changes in addons

### PR DESCRIPTION
**Pull request by karun-singh**
- Added cluster-autoscaler for rancher provisioner, which uses [helm chart](https://github.com/kubernetes/autoscaler/tree/master/charts/cluster-autoscaler) for installation.
- Updated ebs storage class to provision encrypted volumes.

### CODE REVIEW

File: K8s-deployment/K8s-cluster/addons/storage/aws/README.md [hunk 1]
```diff
@@ -10,26 +10,22 @@ kubectl apply -f ebs-storageclass.yaml
...
-   This is enabled in rancher server, through RKE manifest.
 
-2.  Install the drivers by following the steps [here](https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/master/docs/install.md#installation-1)
+   This flag can be enabled in rancher server, through advanced cluster configuration.
+
+2.  Install the drivers by following the steps [here](https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/master/docs/install.md#installation)
 3. Deploy the storage class
 ```console
-kubectl apply -f ebs-csi-storageclass.yaml
+kubectl apply -f ebs/ebs-csi-storageclass.yaml
```
Comment (abhi4578) [inline K8s-deployment/K8s-cluster/addons/storage/aws/README.md:19]: isn't this 
```kubectl apply -f ebs-csi-storageclass.yaml```

File: K8s-deployment/K8s-cluster/addons/cluster-autoscaler/rancher/README.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+## Configuration
+
+The `cluster-autoscaler` for Rancher needs a configuration file to work by
+using `--cloud-config` parameter. Example configuration file - 
+[config.yaml](./config.yaml).
+
+### Permissions
+
+The Rancher server account provided in the `cloud-config` requires the
```
Comment (abhi4578) [inline K8s-deployment/K8s-cluster/addons/cluster-autoscaler/rancher/README.md:15]: will this work with these permissions/scopes and also can u detail the api key  creation?

**Review by abhi4578 — CHANGES_REQUESTED** [review]

File: K8s-deployment/K8s-cluster/addons/storage/aws/README.md (hunk 1 above)
Comment (karun-singh) [reply K8s-deployment/K8s-cluster/addons/storage/aws/README.md]: Fixed

File: K8s-deployment/K8s-cluster/addons/cluster-autoscaler/rancher/README.md (hunk 2 above)
Comment (karun-singh) [reply K8s-deployment/K8s-cluster/addons/cluster-autoscaler/rancher/README.md]: Updated README

**Review by abhi4578 — APPROVED** [review]

**Pull request merged**